include/inexor/vulkan-renderer/io/octree_parser.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "inexor/vulkan-renderer/io/byte_stream.hpp"
+#include "inexor/vulkan-renderer/world/cube.hpp"
+
+#include <memory>
+#include <utility>
+
+namespace inexor::vulkan_renderer::io {
+constexpr std::uint32_t LATEST_OCTREE_FORMAT = 0;
+
+/// Serialization of an octree.
+[[nodiscard]] ByteStream serialize_octree(const std::shared_ptr<const world::Cube> cube,
+                                          const std::uint32_t version = LATEST_OCTREE_FORMAT);
+[[nodiscard]] std::shared_ptr<world::Cube> deserialize_octree(const ByteStream &stream);
+
+/// Specific version serialization.
+template <std::size_t version>
+[[nodiscard]] ByteStream serialize_octree_impl(const std::shared_ptr<const world::Cube> cube);
```

Comment:
Not sure if `version` should be a template parameter. This works fine currently, but if we ever want to read the version from a file at runtime this will not work.

Reply:
It is just used to have a specific version implemented, it should not be used on it's own. There is `serialize_octree` and `deserialize_octree`.